### PR TITLE
added authorization filter

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

**Review by tarkaiev — APPROVED** [review]
well done!

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        protectedUrls.put("/user/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/cart/products", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/cart/products/add", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/cart/products/delete", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/complete", Set.of(Role.RoleName.USER));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:38]: user orders?

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        if (isAuthorized(user, protectedUrls.get(url))) {
+            filterChain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/views/accessDenied.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:38]: added

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**